brave/browser-laptop
# Fixes extra visit time on first logged visit being ignored

Fixes  #13993

**This is a work in progress while we decide the best approach to fix this, un-needed appActions will be removed if this approach is continued with.**

The problem occurs as a result of #13900 . `pageDataChanged` is being dispatched via the PublisherToggle component once the time/visits to include a publisher in the ledger have been satisfied, effectively updating the `isVisibleInLedger` property. The problem is this is done right when the minimum time to log visit is reached, meaning an extra time on that visit is cut off.

We have essentially created via #13900 what seems to be a pretty effective method of determining when a publisher is going to be added to the ledger, so this approach removes the dispatch for `pageDataChanged` and updates the visibility status in the UI. This is confirmed via `pageDataChanged` when the window is blurred/tabs are switched and loaded, etc. This preserves the publisher toggle update without interfering with logging visit time.

The next steps are worth discussing as this approach may have unintended implications.

cc: @NejcZdovc @srirambv 

## Submitter Checklist:


## Test Plan:

  1. Clean install Brave from this patch
  2. Enable payments 
  3. Set minimums to 5 seconds/1 visit
  4. Visit different sites in new tabs and spend different amounts of time on each > 5 seconds.
  5. Confirm that each publisher has the expected amount of visit time in the ledger.

## Reviewer Checklist:


Tests
